src/sidebar/store/modules/selection.js

```diff
@@ -138,22 +149,32 @@ const update = {
     const tabSettings = setTab(state.selectedTab, selectedTab);
     return {
       filterQuery: null,
-      forceVisible: {},
-      selectedAnnotationMap: null,
+      forcedVisible: {},
+      selected: {},
       ...tabSettings,
     };
   },
 
   CLEAR_SELECTED_ANNOTATIONS: function () {
-    return { filterQuery: null, selectedAnnotationMap: null };
+    return { filterQuery: null, selected: {} };
   },
 
   SELECT_ANNOTATIONS: function (state, action) {
-    return { selectedAnnotationMap: action.selection };
+    return { selected: action.selection };
+  },
+
+  TOGGLE_SELECTED_ANNOTATIONS: function (state, action) {
+    const selection = { ...state.selected };
+    action.toggleIds.forEach(id => {
+      selection[id] = !selection[id];
+    });
+    return { selected: selection };
   },
 
   FOCUS_ANNOTATIONS: function (state, action) {
-    return { focusedAnnotations: [...action.focusedTags] };
+    const focused = {};
+    action.focusedTags.forEach(tag => (focused[tag] = true));
```

Comment:
The logic of turning a list of strings into an `Object.<string,true>` set is repeated a few times. This could perhaps be moved to a helper.

Reply:
Thanks, did it.